### PR TITLE
[@types/glob-stream] Update Declarations According to `glob-stream` v8

### PR DESCRIPTION
**Pull request by neurolag**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/gulpjs/glob-stream/releases/tag/v8.0.0>, <https://github.com/gulpjs/glob-stream/tree/v8.0.0#glob-stream>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Related to #65258
As mentioned by @jakebailey, `glob-stream` got a new major release only recently.
Changes made in PR address this major update.

### CODE REVIEW

File: types/glob-stream/index.d.ts [hunk 1]
```diff
@@ -7,33 +7,42 @@
...
         base: string;
         path: string;
     }
 
-    export type UniqueByStringPredicate = 'cwd' | 'base' | 'path';
+    export type UniqueByStringPredicate = keyof Entry;
     export type UniqueByFunctionPredicate = (entry: Entry) => string;
 
-    export interface Options extends glob.GlobOptions {
+    export interface Options extends PicomatchOptions {
```
Comment (jakebailey) [inline types/glob-stream/index.d.ts:26]: Do you mind leaving a comment here regarding why this is pictomatch and not anymatch? e.g.

> `anymatch` passes options down to `picomatch`. We use `pictomatch`'s options here as `anymatch`'s types are hand-written and incorrectly describe the set of possible options that are passed through.

Comment (manuth) [reply types/glob-stream/index.d.ts:29]: Alright! Should be up now
Thanks for taking a look

Comment (manuth) [reply types/glob-stream/index.d.ts:29]: Alright! Should be up now
Thanks for taking a look

**Review by jakebailey — APPROVED** [review]

**Comment by manuth** [comment]
Ready to merge

**Pull request merged**